bump hack/tools dependencies

tuminoid opened the pull request:
Bumping some indirect dependencies to get rid of most public vulnerabilities in said components. sigstore/cosign I'm skipping the uplift as bumping it breaks prowimagebuilder.

Following indirect dependencies are bumped, and all of the other changes are from tidy.

containerd 1.5.9 removed:
- https://osv.dev/vulnerability/GHSA-crp2-qrr5-8pq7
- https://osv.dev/vulnerability/GHSA-5ffw-gxpp-mxpf
- https://osv.dev/vulnerability/GHSA-2qjp-425j-52j9
- https://osv.dev/vulnerability/GHSA-259w-8hf6-59c2
- https://osv.dev/vulnerability/GHSA-hmfx-3pcx-653p

x/net 0.0.0-20220722155237-a158d28d115b to 0.8.0:
- https://osv.dev/vulnerability/GHSA-69cg-p879-7622
- https://osv.dev/vulnerability/GHSA-fxg5-wq6x-vr4w
- https://osv.dev/vulnerability/GO-2022-1144
- https://osv.dev/vulnerability/GHSA-vvpx-j8f3-3w6h

theupdateframework/go-tuf 0.0.0-20210722233521-90e262754396 to 0.3.2:
- https://osv.dev/vulnerability/GHSA-66x3-6cw3-v5gj
- https://osv.dev/vulnerability/GHSA-3633-5h82-39pq

docker/distribution 2.7.1+incompatible to 2.8.1+incompatible:
- https://osv.dev/vulnerability/GHSA-qq97-vm5h-rrhg

docker/docker 20.10.13+incompatible to 20.10.24+incompatible:
- https://osv.dev/vulnerability/GHSA-232p-vwff-86mp
- https://osv.dev/vulnerability/GHSA-33pg-m6jh-5237
- https://osv.dev/vulnerability/GHSA-6wrf-mxfj-pf5p

chases2 commented:
/ok-to-test

tuminoid commented:
> Test name	Commit	Details	Required	Rerun command
> pull-test-infra-verify-lint	[26d350d](https://github.com/kubernetes/test-infra/commit/26d350d60010f823ff958c5b5870747c0e64dace)	[link](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/29305/pull-test-infra-verify-lint/1646925296026259456)	true	`/test pull-test-infra-verify-lint`

Failure in go-lint is due loopclosure issue already existing in test code in master. Fixing that in separate PR: https://github.com/kubernetes/test-infra/pull/29305

tuminoid commented:
/test pull-test-infra-verify-lint

Loopclosure fix PR hs been merged, lets see if this passes without rebase.

tuminoid commented:
/cc @furkatgofurov7 @stevekuznetsov

tuminoid commented:
Rebased, which changed content also a bit. Revised description.

BenTheElder commented on prow/plugins/trick-or-treat/trick-or-treat_test.go:
```diff
@@ -45,15 +45,13 @@ func TestImages(t *testing.T) {
 			t.Parallel()
 			for i := 0; i < 3; i++ {
-				// copy to scoped variable, get rid of loopclosure linting error
-				imgURL := imgURL
```
so while the current version looks wrong, removing this doesn't seem right either

We should be doing:

```
for i := range candiesImges {
     imgURL := candiesImges[i]
 ```

Or similar.

This needs to happen _outside_ the `func(t *testing.T)`, right now this code is bugged in both versions AFAICT

tuminoid replied:
Removed the prow changes from this PR. See https://github.com/kubernetes/test-infra/pull/29305#issuecomment-1517277233

BenTheElder commented on prow/plugins/trick-or-treat/trick-or-treat_test.go:
```diff
@@ -45,15 +45,13 @@ func TestImages(t *testing.T) {
-				// copy to scoped variable, get rid of loopclosure linting error
-				imgURL := imgURL
-				toobig, err := github.ImageTooBig(imgURL)
+				toobig, err := github.ImageTooBig(imgURL) //nolint:govet
```
and then these nolint comments should be removed

tuminoid commented:
Thanks for the review @BenTheElder.

You're correct on the fix, but I'm trying to contain the PR to just bump the deps, and that `go-lint` complaint about loopclosure is pre-existing issue. `make go-lint` fails on master right now.

Fixing it as you suggested makes the test actually work, and exposes that TestImages tests aren't really working (10 failures) since the images it tests against are gone, ie. I'd need to find new images to test against, which is quite unrelated to the deps bump. Hence I leaned towards ignoring the complaint in this PR.

BenTheElder commented:
I see.

Given these are in the tools module which is only used at development time and not distributed to users, I'd actually say fixing linter errors is higher priority than e.g. containerd vulnerabilities that we are in no way actually exposed to. We're not running containerd here.

we should not have failing make targets. That's worth fixing and making sure CI is covering.

BenTheElder commented:
> make go-lint fails on master right now.

That doesn't seem to be the case?

```shell
$ git rev-parse HEAD
ec4335b54bc5c6f53a56c6ac7db148fe6d495c55

$ make go-lint
hack/make-rules/verify/golangci-lint.sh
Ensuring go version.
Install golangci-lint.
Run golangci-lint.
Go version: go version go1.18 darwin/amd64
Golangci-lint version: golangci-lint has version v1.49.0 built from (unknown, mod sum: "h1:I8WHOavragDttlLHtSraHn/h39C+R60bEQ5NoGcHQr8=") on (unknown)

$ echo $?
0
```

tuminoid commented:
> > make go-lint fails on master right now.
> 
> That doesn't seem to be the case?
> 
> ```shell
> $ git rev-parse HEAD
> ec4335b54bc5c6f53a56c6ac7db148fe6d495c55
> 
> $ make go-lint
> hack/make-rules/verify/golangci-lint.sh
> Ensuring go version.
> Install golangci-lint.
> Run golangci-lint.
> Go version: go version go1.18 darwin/amd64
> Golangci-lint version: golangci-lint has version v1.49.0 built from (unknown, mod sum: "h1:I8WHOavragDttlLHtSraHn/h39C+R60bEQ5NoGcHQr8=") on (unknown)
> 
> $ echo $?
> 0
> ```

That's weird as it does fail for me with the same steps.

```shell
❯ git rev-parse HEAD
ec4335b54bc5c6f53a56c6ac7db148fe6d495c55

❯ make go-lint
hack/make-rules/verify/golangci-lint.sh
Ensuring go version.
Install golangci-lint.
Run golangci-lint.
Go version: go version go1.18 linux/amd64
Golangci-lint version: golangci-lint has version v1.49.0 built from (unknown, mod sum: "h1:I8WHOavragDttlLHtSraHn/h39C+R60bEQ5NoGcHQr8=") on (unknown)
prow/plugins/trick-or-treat/trick-or-treat_test.go:49:15: loopclosure: loop variable imgURL captured by func literal (govet)
				imgURL := imgURL
				          ^
ERROR: golangci-lint failed
make: *** [Makefile:72: go-lint] Error 1
```

tuminoid commented:
Removed the prow test fix and as expected, it is failing to loopclosure again. Anyways, I'm going to leave this PR as is, and any fix to loopclosures/tests is going to be in another PR, https://github.com/kubernetes/test-infra/pull/29342 or some new one.

tuminoid commented:
> Removed the prow test fix and as expected, it is failing to loopclosure again. Anyways, I'm going to leave this PR as is, and any fix to loopclosures/tests is going to be in another PR, #29342 or some new one.

This PR will be clean after #29342 is merged. I'll rebase when it is merged to get tests green.

tuminoid commented:
> This PR will be clean after #29342 is merged. I'll rebase when it is merged to get tests green.

#29342 was merged, and this PR rebased, and now the tests should pass.

aojea commented:
/lgtm
/approve

tuminoid commented:
Since last test run, #29415 had reverted one of the loopclosure fixes made in #29342 in `repoowners_test.go`,  hence the linting failure. Rebased this PR, and added the fix back, and tests are passing again. Can you re-LGTM @aojea ?

aojea commented:
/lgtm
